From 1e64bc4c4379ae3e55be3c518e0e3e22c8c4ae77 Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Tue, 14 Nov 2017 13:23:57 +0200
Subject: [PATCH] ntp: fix build with musl libc

The configure script enables HAVE_LINUX_TIMESTAMPING_OPT_PKTINFO based
on the existence of struct scm_ts_pktinfo and
SOF_TIMESTAMPING_OPT_PKTINFO that were introduced in Linux kernel v4.13
in the linux/net_tstamp.h kernel header. But this feature also requires
SCM_TIMESTAMPING_PKTINFO that is defined in the socket.h header. musl
libc provides its own version of socket.h, and as of musl version 1.1.16
this macro is missing. Define this macro in the code as a temporary
measure until musl is updated to its latest version.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: not upstreamable
---
 ntp_io_linux.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ntp_io_linux.c b/ntp_io_linux.c
index 00caed06f7ba..2214a6f6f6c5 100644
--- a/ntp_io_linux.c
+++ b/ntp_io_linux.c
@@ -574,6 +574,10 @@ extract_udp_data(unsigned char *msg, NTP_Remote_Address *remote_addr, int len)
 
 /* ================================================== */
 
+#ifndef SCM_TIMESTAMPING_PKTINFO
+#define SCM_TIMESTAMPING_PKTINFO 58
+#endif
+
 int
 NIO_Linux_ProcessMessage(NTP_Remote_Address *remote_addr, NTP_Local_Address *local_addr,
                          NTP_Local_Timestamp *local_ts, struct msghdr *hdr, int length)
-- 
2.15.0

